R/process-dir.R

```diff
@@ -0,0 +1,260 @@
+  if (fs::file_exists(possible_layout_path)) {
+    return(possible_layout_path)
+  } else {
+    stop(
+      paste0("Layout file for a file ", plate_filepath, " could not be found.")
+    )
+  }
+}
+
+#' @title
+#' Identify if a file is a MBR data file
+#'
+#' @import fs
+#' @importFrom stringr str_split
+#'
+#' @keywords internal
+#'
+is_mbr_data_file <- function(filepath) {
+  format_pattern <- "xpontent|xponent|intelliflex"
+  extension_pattern <- "\\.(xlsx|csv)$"
+  output_pattern <- "RAU|nMFI"
+  layout_pattern <- "_layout"
+
+  stopifnot(fs::file_exists(filepath))
+  filename <- fs::path_file(filepath)
+  filename_splitted <- stringr::str_split(filename, "\\.")
+  basename <- filename_splitted[[1]][1]
+
+  # plate filename has to contain supported format
+  if (!grepl(format_pattern, filename, ignore.case = TRUE)) {
+    return(FALSE)
+  }
+
+  # plate filename extensions has to be supported
+  if (!grepl(extension_pattern, filename)) {
+    return(FALSE)
+  }
+
+  # plate filename has not to contain layout pattern
+  if (grepl(layout_pattern, basename, fixed = TRUE)) {
+    return(FALSE)
+  }
+
+  # plate filename has not to contain supported output format
+  # as not to mix it up with output files
+  if (grepl(output_pattern, basename, ignore.case = TRUE)) {
+    return(FALSE)
+  }
+
+  return(TRUE)
+}
+
+#' @title
+#' Try to detect the format of a file
+#'
+#' @import fs
+#' @importFrom stringr str_split
+#'
+#' @keywords internal
+#'
+detect_mbr_format <- function(filepath, format = NULL) {
+  if (!is.null(format)) {
+    stopifnot(format %in% c("xPONENT", "INTELLIFLEX"))
+    return(format)
+  }
+
+  stopifnot(fs::file_exists(filepath))
+  filename <- fs::path_file(filepath)
+  filename_splitted <- stringr::str_split(filename, "\\.")
+  basename <- filename_splitted[[1]][1]
+
+  xponent_pattern <- "(xpontent|xponent)"
+  intelliflex_format <- "(intelliflex)"
+  if (grepl(xponent_pattern, basename, ignore.case = TRUE)) {
+    return("xPONENT")
+  } else if (grepl(intelliflex_format, basename, ignore.case = TRUE)) {
+    return("INTELLIFLEX")
+  } else {
+    stop("The format of the file could not be detected.")
+  }
+}
+
+#' @title
+#' Get output directory for a given input file
+#'
+#' @import fs
+#'
+#' @keywords internal
+#'
+get_output_dir <- function(
+    input_file,
+    input_dir,
+    output_dir = NULL,
+    flatten_output = FALSE) {
+  output_root <- ifelse(is.null(output_dir), input_dir, output_dir)
+  if (!fs::dir_exists(output_root)) {
+    stop("Output directory does not exist.")
+  }
+  if (flatten_output) {
+    current_output_dir <- output_root
+  } else {
+    input_file_rel_path <- fs::path_rel(input_file, input_dir)
+    current_output_dir <- fs::path_dir(
+      fs::path_join(c(output_root, input_file_rel_path))
+    )
+  }
+  return(fs::path(current_output_dir))
+}
+
+#' @title
+#' Process a dir of files to generate normalised data and reports
```

Comment:
I would extend this description with additional 2-3 lines (chat gpt could even help)